java/libs/src/main/java/org/physical_web/collection/PhysicalWebCollection.java

```diff
@@ -72,6 +72,15 @@ public UrlDevice getUrlDeviceById(String id) {
   }
 
   /**
+   * Fetches cached URL metadata using the URL broadcasted by the Physical Web device.
+   * @param broadcastUrl The URL broadcasted by the device.
+   * @return Cached metadata relevant to the given URL.
+   */
+  public PwsResult getMetadataByBroadcastUrl(String broadcastUrl) {
+    return mBroadcastUrlToPwsResultMap.get(broadcastUrl);
```

Comment:
...so this should look up resolved url for the given broadcast url and return the result for that.

If we don't have a mapping yet, we need to create one with a call to PWS.

We may also still not have one after call to PWS.

Should this return an async, given that it is the public interface?